### PR TITLE
Added support for tokens

### PR DESCRIPTION
**Pull request by pbalexlear**
Added support for TokenProvider as an authentication method for service bus

### CODE REVIEW

File: Pat.Subscriber/AzureServiceBusMessageReceiverFactory.cs [hunk 1]
```diff
@@ -15,5 +16,12 @@ protected override IMessageReceiver CreateMessageReceiver(string connectionStrin
             return new MessageReceiver(connectionString,
                 EntityNameHelper.FormatSubscriptionPath(topicName, subscriberName));
         }
```
Comment (Bigtalljosh) [inline Pat.Subscriber/AzureServiceBusMessageReceiverFactory.cs]: New line between these please, let's keep it neat :)

File: Pat.Subscriber/MessageReceiverFactory.cs [hunk 2]
```diff
@@ -43,8 +46,18 @@ private List<IMessageReceiver> CreateMessageReceivers()
                 if (!string.IsNullOrEmpty(connectionString))
                 {
                     _log.LogInformation($"Adding on subscription client {clientIndex} to list of source subscriptions");
-                    messageReceivers.Add(CreateMessageReceiver(connectionString,
-                        _config.EffectiveTopicName, _config.SubscriberName));
+                    if (_config.TokenProvider !=null)
```
Comment (Bigtalljosh) [inline Pat.Subscriber/MessageReceiverFactory.cs]: space after `!=`

File: Pat.Subscriber/MessageReceiverFactory.cs [hunk 3]
```diff
@@ -43,8 +46,18 @@ private List<IMessageReceiver> CreateMessageReceivers()
...
+                    {
+
```
Comment (Bigtalljosh) [inline Pat.Subscriber/MessageReceiverFactory.cs]: Please remove random new lines

File: Pat.Subscriber/SubscriptionBuilder.cs [hunk 4]
```diff
@@ -56,8 +58,19 @@ private async Task BuildSubscription(string connectionString, string[] messagesT
         {
             var topicName = _config.EffectiveTopicName;
 
-            var client = new SubscriptionClient(connectionString, topicName, _config.SubscriberName);
-         
+            SubscriptionClient client;
+            if (_config.TokenProvider != null)
+            {
+                ServiceBusConnectionStringBuilder builder = new ServiceBusConnectionStringBuilder(connectionString);
+                Console.WriteLine(builder.Endpoint);
```
Comment (Bigtalljosh) [inline Pat.Subscriber/SubscriptionBuilder.cs]: Did we mean to leave the console log in?

File: Pat.Subscriber/SubscriptionBuilder.cs [hunk 5]
```diff
@@ -56,8 +58,19 @@ private async Task BuildSubscription(string connectionString, string[] messagesT
...
+                client = new SubscriptionClient(builder.Endpoint, topicName, _config.SubscriberName, _config.TokenProvider);
+            }
+            else {
```
Comment (Bigtalljosh) [inline Pat.Subscriber/SubscriptionBuilder.cs]: please new line the brace and remove the empty new line below

File: Pat.Subscriber/AzureServiceBusMessageReceiverFactory.cs (hunk 1 above)
Comment (pbalexlear) [reply Pat.Subscriber/AzureServiceBusMessageReceiverFactory.cs]: new line where? between the params?

File: Pat.Subscriber/SubscriptionBuilder.cs (hunk 4 above)
Comment (pbalexlear) [reply Pat.Subscriber/SubscriptionBuilder.cs]: nope have removed

**Pull request merged**